### PR TITLE
fix: OAuth disconnect

### PR DESCRIPTION
**Pull request by jonas-martinez**
## Description of the changes

If there is no access token in local storage, the access token in OAuthModel should not act as an authentication method.

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [x] I made my own code-review before requesting one

### I included unit tests that cover my changes
  - [ ] 👍 yes
  - [x] 🙅 no, because they aren't needed
  - [ ] 🙋 no, because I need help

### I added/updated the documentation about my changes

- [ ] 📜 README.md
- [ ] 📕 docs/*.md
- [ ] 📓 docs.lenra.io
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by taorepoara** [comment]
I've made a PR to really disconnect the user that should be added to this PR: https://github.com/lenra-io/client-common/pull/96

**Comment by jonas-martinez** [comment]
See https://github.com/ory/hydra/issues/1754#issuecomment-1727406218 for a potential fix

**Comment by taorepoara** [comment]
Waiting for https://github.com/lenra-io/client-common/pull/105

**Comment by jonas-martinez** [comment]
I tested this on the main page (store page) of the client but I did NOT test it on an application. 
We might need to add some documentation to help test applications locally because it became really tiring to setup an app. 
If we could maybe add a mix task to automatically create an app in the database with the proper data and already deployed. We could use an already working deployment from git for that.

File: lib/views/store_page/logout_button.dart [hunk 1]
```diff
@@ -13,9 +13,9 @@ class LogoutButton extends StatelessWidget {
   Widget build(BuildContext context) {
     return ElevatedButton(
       onPressed: () async {
-        await context.read<OAuthModel>().helper.disconnect();
+        await context.read<OAuthModel>().logout();
         // ignore: use_build_context_synchronously
-        context.go("/");
+        context.go("/oauth");
```
Comment (taorepoara) [inline lib/views/store_page/logout_button.dart]: Do not redirect to oauth

**Review by taorepoara — APPROVED** [review]
Seems OK

**Pull request merged**